src/Firehose.Web/Infrastructure/NewCombinedFeedSource.cs

```diff
@@ -191,6 +193,85 @@ DateTimeOffset GetMaxTime(SyndicationItem item)
             return feed;
         }
 
+        private void AddTwitterHandlesToFeed(SyndicationFeed feed)
+        {
+            foreach (var item in feed.Items)
+            {
+                var author = Tamarins.FirstOrDefault(a => MatchesAuthorUrls(a, item.Links.Select(l => l.Uri), item));
+                if (!string.IsNullOrWhiteSpace(item.Title.Type) && author != null && !string.IsNullOrEmpty(author.TwitterHandle))
+                {
+                    item.Title = new TextSyndicationContent(item.Title.Text + " @" + author.TwitterHandle, GetTextKind(item.Title.Type));
+                }
+                else if (author != null && !string.IsNullOrEmpty(author.TwitterHandle))
+                {
+                    item.Title = new TextSyndicationContent(item.Title.Text + " @" + author.TwitterHandle);
+                }
+            }
+        }
+
+        private bool MatchesAuthorUrls(IAmACommunityMember author, IEnumerable<Uri> urls, SyndicationItem item)
+        {
+            var authorHosts = author.FeedUris.Select(au => au.Host.ToLowerInvariant()).Concat(new[] { author.WebSite.Host.ToLowerInvariant() }).ToArray();
+            var feedBurnerAuthors = author.FeedUris.Where(au => au.Host.Contains("feeds.feedburner")).ToList();
+            var mediumAuthors = author.FeedUris.Where(au => au.Host.Contains("medium.com")).ToList();
+            var youtubeAuthors = author.FeedUris.Where(au => au.Host.Contains("youtube.com")).ToList();
+
+            foreach (var itemUrl in urls)
+            {
+                var host = itemUrl.Host.ToLowerInvariant();
+
+                if (host.Contains("medium.com"))
```

Comment:
Some pattern matching and splitting out into separate methods would do very good here